### PR TITLE
New unpack 'Treat as video' option to treat RARs as video files

### PR DESCRIPTION
**Pull request by miigotu**
Quick n dirty config only option for allowing rar files in the show folder without unpacking them, pp will just move them if enabled and unpack is off

Fixes #

Proposed changes in this pull request:
-
-
-

- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/SickRage/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

**Comment by miigotu** [comment]
@sharkykh mind adding html and code for changing this value from the GUI? If DONT_UNPACK is enabled then UNPACK should be automatically disabled when changing the setting
I really need a break lol

**Comment by sharkykh** [comment]
1. Wait, why are you adding another setting if we already have UNPACK?
2. Doesn't make sense, Markup-language-wise, to do checkboxes that disable each other... You could just do that with radio buttons... Something like
   - 0 = Unpacking disabled (ignore rar files)
   - 1 = Unpacking enabled (extract rar files and process the media files they contain)
   - 2 = Move rars to show folder (do not extract)

I would only do that tomorrow anyway, I think

**Comment by miigotu** [comment]
It was really just a quick PoC test for someone on irc, It can be adjusted however necessary.

**Comment by sharkykh** [comment]
Wait, did I get this right?
This option would allow the user to move/copy/symlink/hardlink rar files, as-is, from TV Download dir to be _**in the show's folder**_? like, next to the video after post-processing? :confused: 

I don't think this looks good...

image

**Comment by miigotu** [comment]
Yes, it renames and moves the rar to the show folder, just as if it were a video file itself. Many modern media players can play videos in rars directly, so it saves the user drive space

**Comment by sharkykh** [comment]
Still working on the GUI, didn't touch any python just yet.
What do you think? Better?
image

**Comment by miigotu** [comment]
Sure, I also need to find a better way to auto-find unrar, but thats a different story.

**Comment by sharkykh** [comment]
I'm gonna change `is_media_file` back because I think it'll cause a chain of issues if someone actually decides to use this option. This function gets used in many modules...
Just wanted to let you know. I know what you said...
>  It can be adjusted however necessary.

**Comment by miigotu** [comment]
You dont need to change it back. If the option is enabled then it IS a media file. Thats the point.

**Comment by miigotu** [comment]
I mean, ofc it CAN be done another way, but this is the simplest way and everything will work just fine with how it is too.

**Comment by sharkykh** [comment]
OK, I you're positive it's gonna work then I'll keep it as it is.

**Comment by sharkykh** [comment]
erm... [What about that?](https://github.com/SickRage/SickRage/blob/b0cdd5788e894c90ecc31089947eaf225b447037/sickrage/helper/media_info.py#L119)

**Comment by miigotu** [comment]
It returns (None, None), thats the only method `qualityFromFileMeta` that actually reads the media file, and it returns gracefully.

**Comment by sharkykh** [comment]
Uh, can't push to the branch

**Comment by miigotu** [comment]
Make a PR to this branch ^

**Comment by miigotu** [comment]
Promoted you to a collaborator so you can push to this remote.
Be careful not to push tags, and don;t push direct to develop/master

**Comment by sharkykh** [comment]
```
remote: Permission to SickRage/SickRage.git denied to sharkykh.
fatal: unable to access 'https://github.com/SickRage/SickRage.git/': The requested URL returned error: 403
Pushing to https://github.com/SickRage/SickRage.git
```
Still denied (guess it hasn't updated yet)
I'll open a PR now (had some local issues with my commits till now..)

**Comment by sharkykh** [comment]
I thought it would copy the commit notes too, sorry.
> - Change UNPACK from bool to int,
Implement the 'Treat as video' option for archives
> - Fix add_site_message to support translations

File: gui/slick/views/config_postProcessing.mako [hunk 1]
```diff
@@ -272,22 +272,36 @@
...
+                                        ${_('What to do with archived releases found in your <i>TV Download Dir</i>?')}
+                                    </label>
+                                </div>
+
+                                <div class="col-lg-9 col-md-9 col-sm-8 col-xs-12 pull-right component-desc">
                                     <div class="row">
                                         <div class="col-md-12">
-                                            <input id="unpack" class="enabler" type="checkbox" name="unpack"
-                                                ${('', 'checked="checked"')[bool(sickbeard.UNPACK)]} />
-                                            <label for="unpack">${_('unpack any TV releases in your <i>TV Download Dir</i>?')}</label>
```
Comment (miigotu) [inline gui/slick/views/config_postProcessing.mako:279]: Need to have a label attached to the input. If it needs to be hidden use class `sr-only` on the label

**Review by miigotu — COMMENTED** [review]
Also, need to add a config migration, so that user's existing setting is not lost in their config just by updating.

File: gui/slick/views/config_postProcessing.mako (hunk 1 above)
Comment (sharkykh) [reply gui/slick/views/config_postProcessing.mako:279]: [This is the label ](https://github.com/SickRage/SickRage/pull/3271/files/4e03c68c5f7cb72ee3981a8241587b5eff995fd7#diff-b7e36dabf5283a72bc76bbb6d7c258c5R275). I'll just add a for="unpack" to it.

**Comment by sharkykh** [comment]
About migrating the config there's no need.

Original UNPACK | Config read | New UNPACK
--------- | ------ | ---
False | check_setting_int(0) | 0
True | check_setting_int(1) | 1

It worked for me, like it should because of:
[check_setting_int](https://github.com/SickRage/SickRage/blob/b0cdd5788e894c90ecc31089947eaf225b447037/sickbeard/config.py#L577-L580)

File: gui/slick/views/config_postProcessing.mako (hunk 1 above)
Comment (miigotu) [reply gui/slick/views/config_postProcessing.mako:279]: Yep, just has to be linked with `for=` =P

**Comment by miigotu** [comment]
Maybe that one should also add a site_message if not sickbeard.DEVELOPER and no providers are enabled for the mode

**Comment by sharkykh** [comment]
I'll try and add that

**Comment by sharkykh** [comment]
This will show up only one time: Startup. I didn't think putting it in every search cycle is smart...
It changes according to whatever needs attention.
Link is to /config/providers ofc
I'd love your feedback on my code.

image

**Comment by miigotu** [comment]
"Adjust enabled providers backlog/daily check site_message code. When indexing a 2 item tuple the index value should only be 0, 1,(min 0, max len(tuple) - 1), True, or False. Anything else will result in an IndexError. Also, break out of the loop once we know we have both, to save a little time starting up. Auto-(add|remove) the message when saveProviders"

**Pull request merged**